ims/einstein/operations.py

```diff
@@ -635,3 +639,31 @@ def remake_mappings(self):
             logger.exception('')
         except NotImplementedError:
             pass
+
+    @log
+    def run_script(self, img, script):
+        try:
+            if not self.is_admin:
+                raise AuthorizationFailedException()
+            directory = "/tmp/script/"
+            os.makedirs(directory)
+            script_path = directory + "/test_script.sh"
+            ret = self.fs.map(img)
+            decode_script = base64.b64decode(script)
+            with open(script_path, "w") as write_script:
+                write_script.write(decode_script)
+            os.chmod(script_path, S_IRWXU)
+            p = subprocess.Popen(script_path + " " + ret, shell=True,
```

Comment:
The design we decided on with the Secure Cloud team has their script performing the actual mounting of the image. I will change it to clone the image before mapping and delete it after.